### PR TITLE
Save tts setting values( enabletts,rate,volume,voice) into persistent

### PR DESCRIPTION
**Pull request by vdinak240**
Save tts setting values( enabletts,rate,volume,voice) into persistent

### CODE REVIEW

File: TextToSpeech/impl/TTSManager.cpp [hunk 1]
```diff
@@ -50,19 +49,20 @@ TTSManager::~TTSManager() {
...
-    }
-
+    bool updated = false;
+    updated |= m_defaultConfiguration.setEnabled(enable);
+    TTSLOG_INFO("TTS is %s", enable ? "Enabled" : "Disabled");
+    if(updated)
+        m_defaultConfiguration.updateConfigStore();
+    if(!m_defaultConfiguration.enabled())
+        shut(0);
+    m_callback->onTTSStateChanged(m_defaultConfiguration.enabled());
```
Comment (varumugam123) [inline TextToSpeech/impl/TTSManager.cpp:59]: onTTSStateChanged() & ensurePipeline() should happen only if there is a state change

File: TextToSpeech/TextToSpeechImplementation.cpp [hunk 2]
```diff
@@ -503,5 +504,44 @@ namespace Plugin {
...
     }
 
+    bool _readFromFile(std::string filename, TTS::TTSConfiguration &ttsConfig)
+    {
+        Core::File file(filename);
+        if(file.Open()) {
+            JsonObject config;
+            if(config.IElement::FromFile(file)) {
+            Core::JSON::Boolean enabled = config.Get("enabled").Boolean();
+            ttsConfig.setEnabled(enabled.Value());
```
Comment (varumugam123) [inline TextToSpeech/TextToSpeechImplementation.cpp:514]: Should we check the existence of "enable" key in the config before updating?

File: TextToSpeech/TextToSpeechImplementation.cpp [hunk 3]
```diff
@@ -503,5 +504,44 @@ namespace Plugin {
...
+            ttsConfig.setVolume(std::stod(GET_STR(config,"volume","0.0")));
+            ttsConfig.setRate(static_cast<uint8_t>(std::stoi(GET_STR(config,"rate","0"))));
+            ttsConfig.setVoice(GET_STR(config,"voice",""));
+            ttsConfig.setLanguage(GET_STR(config,"language",""));
+
+            TTSLOG_INFO("Language : %s", ttsConfig.language().c_str());
```
Comment (varumugam123) [inline TextToSpeech/TextToSpeechImplementation.cpp:520]: Is this logging redundant?

File: TextToSpeech/TextToSpeechImplementation.cpp [hunk 4]
```diff
@@ -82,7 +82,7 @@ namespace Plugin {
         } else {
             TTSLOG_WARNING("Doesn't find default voice configuration");
         }
-
+        ttsConfig->loadFromConfigStore();
         TTSLOG_INFO("TTSEndPoint : %s", ttsConfig->endPoint().c_str());
         TTSLOG_INFO("SecureTTSEndPoint : %s", ttsConfig->secureEndPoint().c_str());
         TTSLOG_INFO("Language : %s", ttsConfig->language().c_str());
```
Comment (varumugam123) [inline TextToSpeech/TextToSpeechImplementation.cpp:88]: Please log the "enabled" state

File: TextToSpeech/impl/TTSManager.cpp [hunk 5]
```diff
@@ -50,19 +49,20 @@ TTSManager::~TTSManager() {
...
 TTS_Error TTSManager::enableTTS(bool enable) {
-
-    if(m_ttsEnabled != enable) {
-        m_ttsEnabled = enable;
-        TTSLOG_INFO("TTS is %s", enable ? "Enabled" : "Disabled");
-        m_callback->onTTSStateChanged(m_ttsEnabled);
-        m_speaker->ensurePipeline(m_ttsEnabled);
-    }
-
+    bool updated = false;
```
Comment (varumugam123) [inline TextToSpeech/impl/TTSManager.cpp:52]: This "updated" variable may not be required

**Review by varumugam123 — CHANGES_REQUESTED** [review]

File: TextToSpeech/TextToSpeechImplementation.cpp (hunk 2 above)
Comment (vdinak240) [reply TextToSpeech/TextToSpeechImplementation.cpp:515]: No need. If the key is not present or the value of key is empty ."False" is returned.

File: TextToSpeech/TextToSpeechImplementation.cpp (hunk 4 above)
Comment (vdinak240) [reply TextToSpeech/TextToSpeechImplementation.cpp:88]: Addressed in the new patchset

**Review by varumugam123 — APPROVED** [review]

**Review by bobseamon — APPROVED** [review]

**Pull request merged**